### ISSUE TITLE
invoice template per project

### ISSUE THREAD

**Issue by BeckeBauer** [kimai/kimai#4042]
### Describe the problem

We have different projects for the same customer which are invoiced on basis of different invoice templates. Right now, a single invoice template can be assigned to a customer:
<img width="668" alt="image" src="https://github.com/kimai/kimai/assets/51003649/6f09b42c-9b8b-414a-9771-31b76c056949">
but not to a project:
![image](https://github.com/kimai/kimai/assets/51003649/a8d7c900-56b8-4f8f-a278-53ed157b5d0f)

### Describe the solution you'd like

It would be great if an invoice template could be set to a project which if set prevails over the template specified for the customer

**Comment by kevinpapst** [comment]
Create multiple customers... ?

**Comment by BeckeBauer** [comment]
> Create multiple customers... ?

Do you mean to create the same customer several times but with different projects? 
We often use similar project names for different customers so that we are used to pick a customer and not a project when creating a timesheet. Assuming I understand your suggestion correctly, it means, one need to go through the list of the same customer name until the project is shown for which the timesheet needs to be recorded. That seem unpractical and error prone...

**Comment by kevinpapst** [comment]
I agree with the "unpractical" part. 
But I am not a fan of duplicating another config there.
If I could, I would remove some of those I added in the past...

If this issue manages to get more votes, I will think about it.

**Comment by LiaraAlis** [comment]
I have a few thoughts on this... I think it makes much more sense if a invoice template depends on the project and not on the customer. Because the project defines which type of invoicing is required and not the customer.
I suggest to move this setting from the customer to the project.

**Comment by kevinpapst** [comment]
> I have a few thoughts on this

Okay, you gave none actually 😁 What are those thoughts?  

> Because the project defines which type of invoicing is required and not the customer.

Usually the customer location (country) defines both target language and tax regulation. 

In your case that might be different, but the general rule is that the customer is relevant for the used template.
I have customers with multiple projects, why should I duplicate the same setting over and over again?

It should be fairly simple, to create a plugin that add this setting to projects. Why not create it and can sell it to the crowd?! I would even help with tipps for the implementation.

**Comment by LiaraAlis** [comment]
> Okay, you gave none actually 😁 What are those thoughts?

Of course, I did already.

> Usually the customer location (country) defines both target language and tax regulation.

That's right. But a project can define more factors, that can determine which template should be used. e.g. because different detailed variants are necessary depending on the project. If this is the case, it is very unpractible to create the same customer multiple times.

> In your case that might be different […]

It's not really my case. I just want to help @BeckeBauer.

**Comment by piyushpaliwal** [comment]
I agree to the idea that invoice templates should be associated to the project instead of customer. That ways, it gives much more granular control, and it solves another use-case.

There are times where you want to have different tax rates for different projects in a single invoice. Currently, there is an assumption that invoices will have a single tax rate; but in reality the tax rates depend on individual item/service within the invoice.

Example - 
Customer contracts us to do 2 services for him.
- A service 'X' could be subsidised by a govt scheme, which would require a GST of say 5%
- A service 'Y' would entail a different GST amount, say 15%'

In this case, when we generate an invoice, there should be 2 line items and tax amount for them calculated at service/project level.
